src/braket/aws/aws_quantum_task_batch.py

```diff
@@ -201,7 +201,7 @@ def _tasks_inputs_gatedefs(
                     batch_length = arg_length
 
         for i, arg_length in enumerate(arg_lengths):
-            if arg_length == 1:
+            if isinstance(args[i], dict) or isinstance(args[i], single_task_type):
```

Comment:
I think you can
```suggestion
            if isinstance(args[i], (dict, single_task_type)):
```

Reply:
Good catch, yep, the union of the tuple should work